scripts/src/saforcertadmin/push_secrets.py

```diff
@@ -0,0 +1,75 @@
+from base64 import b64encode
+from nacl import encoding, public
+import logging
+import os
+import sys
+import requests
+import argparse
+
+token = os.environ.get("GITHUB_TOKEN")
+headers = {'Accept': 'application/vnd.github.v3+json', 'Authorization': f'token {token}'}
+
+logging.basicConfig(level=logging.INFO)
+
+def encrypt(public_key: str, secret_value: str) -> str:
+    """Encrypt a Unicode string using the public key."""
+    public_key = public.PublicKey(public_key.encode("utf-8"), encoding.Base64Encoder())
+    sealed_box = public.SealedBox(public_key)
+    encrypted = sealed_box.encrypt(secret_value.encode("utf-8"))
+    return b64encode(encrypted).decode("utf-8")
+
+def get_repo_public_key(repo):
+    response = requests.get(f'https://api.github.com/repos/{repo}/actions/secrets/public-key', headers=headers)
+    if response.status_code != 200:
+        logging.error(f"unexpected response getting repo public key : {response.status_code} : {response.reason}")
+        sys.exit(1)
+    response_json = response.json()
+    return response_json['key_id'], response_json['key']
+
+def get_repo_secrets(repo):
+    secret_names = []
+    response = requests.get(f'https://api.github.com/repos/{repo}/actions/secrets', headers=headers)
+    if response.status_code != 200:
+        logging.error(f"unexpected response getting secrets : {response.status_code} : {response.reason}")
+        sys.exit(1)
+    response_json = response.json()
+    for i in range(response_json['total_count']):
+        secret_names.append(response_json['secrets'][i]['name'])
+    return secret_names
+
+def create_or_update_repo_secrets(repo, secret_name, key_id, encrypted_value):
+    response = requests.put(f'https://api.github.com/repos/{repo}/actions/secrets/{secret_name}', json={'key_id': key_id, 'encrypted_value': encrypted_value}, headers=headers)
+    if response.status_code != 200 and response.status_code != 204:
```

Comment:
Interesting check for 200 and 204, would other 200 responses be failures? Also would expect PUT to return 201.

Reply:
My bad. Response code will be 201 or 204 as per the API documentation here https://docs.github.com/en/rest/reference/actions#create-or-update-an-organization-secret--code-samples

Reply:
Updated now.

Reply:
Looks good.